### PR TITLE
Various fixes to 0.9.9 rc.3

### LINKED ISSUE #1042: New cluster credentials cannot be rendered
Taken from https://github.com/kubernetes-incubator/kube-aws/issues/1003#issuecomment-347084904. This is still a similar problem in that that render does not work for a new cluster.

Using version 3e402be845b5362756594b6cd61d962aab9f84d6 or v0.9.9-rc.3:

1. Run a basic `init`. The one from the [quick start guide](https://github.com/c-knowles/kube-aws/blob/5ce4fc461f06a6c95e4e407ae3059411e09ac252/docs/tutorials/quick-start.md) is ok.
1. Copy your ca.pem and ca-key.pem to `credentials` folder
1. Run `render credentials`, the following error occurs:

```bash
➜ kube-aws render credentials                                            
Generating credentials...
-> Parsing existing TLS CA
-> Generating new assets
--> Summarizing the configuration
    Kubelet TLS bootstrapping enabled=false, TLS certificates managed by kube-aws=true, CA key required on controller nodes=false
--> Writing to the storage
INFO: Writing 1070 bytes to credentials/ca.pem
INFO: Creating a symlink from worker-ca.pem to ca.pem
Error: Error creating assets: Not sure what to do for credentials/worker-ca-key.pem
```

The same process and ca files work with older versions of kube-aws.

The error message comes from https://github.com/kubernetes-incubator/kube-aws/pull/1014 but not sure why it's hitting that case for worker-ca-key.pem.

### PR DESCRIPTION
**Pull request by mumoshu**
@c-knowles @danielfm This is intended to fix #1042 and #1039, which is already working for my clusters.
Would you mind building & testing this?

### CODE REVIEW

File: core/controlplane/config/templates/cloud-config-controller [hunk 1]
```diff
@@ -789,6 +789,11 @@ write_files:
         sleep 3
       done
 
+      # Work-around for https://github.com/kubernetes-incubator/kube-aws/issues/1039
+      if kubectl --namespace kube-system get apiservice v1beta1.metrics.k8s.io; then
+        kubectl --namespace kube-system delete apiservice v1beta1.metrics.k8s.io
+      fi
```
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: Indeed this looks too naive but I'm not yet sure what's a better option here.

File: core/controlplane/config/templates/cloud-config-controller [hunk 2]
```diff
@@ -2665,46 +2670,35 @@ write_files:
           namespace: kube-system
           labels:
 
-  - path: /srv/kubernetes/manifests/metrics-server-de.yaml
+  - path: /etc/kubernetes/manifests/metrics-server.yaml
     content: |
-        apiVersion: extensions/v1beta1
-        kind: Deployment
```
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: I was unable to make this work w/ Deployment or DaemonSet.
When this was a deployment/daemonset, `kubectl --namespace kube-system get apiservice v1beta1.metrics.k8s.io` and `kubectl --namespace kube-system delete apiservice v1beta1.metrics.k8s.io` seemed to hang forever.

**Comment by mumoshu** [comment]
// cc'ing @camilb for metrics-server

**Comment by c-knowles** [comment]
@mumoshu yes ok, will try this out now.

**Comment by c-knowles** [comment]
I can confirm that it seems https://github.com/kubernetes-incubator/kube-aws/issues/1042 is resolved with this PR and after `render credentials` I have this in the `credentials` folder:
```
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 admin-key.pem
-rw-------  1 cknowles  staff   1.1K Nov 29 11:45 admin.pem
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 apiserver-key.pem
-rw-------  1 cknowles  staff   1.3K Nov 29 11:45 apiserver.pem
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 ca-key.pem
-rw-------  1 cknowles  staff   1.0K Nov 29 11:45 ca.pem
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 etcd-client-key.pem
-rw-------  1 cknowles  staff   1.1K Nov 29 11:45 etcd-client.pem
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 etcd-key.pem
lrwxr-xr-x  1 cknowles  staff     6B Nov 29 11:45 etcd-trusted-ca.pem -> ca.pem
-rw-------  1 cknowles  staff   1.1K Nov 29 11:45 etcd.pem
-rw-------  1 cknowles  staff   344B Nov 29 11:45 kubelet-tls-bootstrap-token
-rw-------  1 cknowles  staff     0B Nov 29 11:45 tokens.csv
lrwxr-xr-x  1 cknowles  staff     6B Nov 29 11:45 worker-ca.pem -> ca.pem
-rw-------  1 cknowles  staff   1.6K Nov 29 11:45 worker-key.pem
-rw-------  1 cknowles  staff   1.1K Nov 29 11:45 worker.pem
```

**Comment by jlk** [comment]
I ran into #1042 as well and this fixed for me as well, thanks!

**Comment by mumoshu** [comment]
Merging this because I've already tested this bunch of times in #1039.

File: core/controlplane/config/templates/cloud-config-controller [hunk 3]
```diff
@@ -781,108 +781,111 @@ write_files:
...
       {{- if .Addons.Rescheduler.Enabled }}
-      kubectl apply -f "${mfdir}/kube-rescheduler-de.yaml"
+      applyall "${mfdir}/kube-rescheduler-de.yaml"
       {{- end }}
 
       # API Services
-      for manifest in {metrics-server,}; do
-          kubectl apply -f "${mfdir}/$manifest-apisvc.yaml"
-      done
+      applyall "${mfdir}/metrics-server-apisvc.yaml"
```
Comment (c-knowles) [inline core/controlplane/config/templates/cloud-config-controller]: @mumoshu there's some inconsistency here - `"${mfdir}"` and `"${mfdir}`. I'm getting this on latest master:
```
Dec 07 08:17:38 ip-10-0-30-104.eu-west-1.compute.internal retry[1983]: /opt/bin/install-kube-system: line 96: unexpected EOF while looking for matching `"'
Dec 07 08:17:38 ip-10-0-30-104.eu-west-1.compute.internal retry[1983]: Attempt 3 failed and there are no more attempts left!
Dec 07 08:17:38 ip-10-0-30-104.eu-west-1.compute.internal retry[1983]: /opt/bin/retry: line 12: return: can only `return' from a function or sourced script
```

Comment (mumoshu) [reply core/controlplane/config/templates/cloud-config-controller]: Probably your issue would be resolved by #1060. Would you mind trying?

Comment (c-knowles) [reply core/controlplane/config/templates/cloud-config-controller]: @mumoshu ok will try that one, thanks.

Comment (c-knowles) [reply core/controlplane/config/templates/cloud-config-controller]: @mumoshu yep that seems to have gotten the kube system deploys working again.

**Pull request merged**